e2e/e2e_ext_test.go

```diff
@@ -2272,6 +2272,171 @@ func TestPopPayouts(t *testing.T) {
 	}
 }
 
+func TestPopPayoutsMultiplePages(t *testing.T) {
+	db, pgUri, sdb, cleanup := createTestDB(context.Background(), t)
+	defer func() {
+		db.Close()
+		sdb.Close()
+		cleanup()
+	}()
+
+	ctx, cancel := defaultTestContext()
+	defer cancel()
+
+	includedL2Keystone := hemi.L2Keystone{
+		Version:            1,
+		L1BlockNumber:      11,
+		L2BlockNumber:      22,
+		ParentEPHash:       fillOutBytes("parentephash", 32),
+		PrevKeystoneEPHash: fillOutBytes("prevkeystoneephash", 32),
+		StateRoot:          fillOutBytes("stateroot", 32),
+		EPHash:             fillOutBytes("ephash", 32),
+	}
+
+	btcHeaderHash := fillOutBytes("btcheaderhash", 32)
+
+	btcBlock := bfgd.BtcBlock{
+		Hash:   btcHeaderHash,
+		Header: fillOutBytes("btcheader", 80),
+		Height: 99,
+	}
+
+	if err := db.BtcBlockInsert(ctx, &btcBlock); err != nil {
+		t.Fatal(err)
+	}
+
+	// insert 151 pop payouts to different miners, get the first 3 pages,
+	// we expect result counts like so : 100, 51, 0
+	var txIndex uint64 = 1
+
+	for range 151 {
+
+		privateKey, err := dcrsecp256k1.GeneratePrivateKeyFromRand(rand.Reader)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		publicKey := privateKey.PubKey()
+		publicKeyUncompressed := publicKey.SerializeUncompressed()
+
+		txIndex++
+		popBasis := bfgd.PopBasis{
+			BtcTxId:             fillOutBytes("btctxid1", 32),
+			BtcRawTx:            []byte("btcrawtx1"),
+			PopTxId:             fillOutBytes("poptxid1", 32),
+			L2KeystoneAbrevHash: hemi.L2KeystoneAbbreviate(includedL2Keystone).Hash(),
+			PopMinerPublicKey:   publicKeyUncompressed,
+			BtcHeaderHash:       btcHeaderHash,
+			BtcTxIndex:          &txIndex,
+		}
+
+		if err := db.PopBasisInsertFull(ctx, &popBasis); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	_, _, bfgWsurl, _ := createBfgServer(ctx, t, pgUri, "", 1)
+
+	_, _, bssWsurl := createBssServer(ctx, t, bfgWsurl)
+
+	c, _, err := websocket.Dial(ctx, bssWsurl, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer c.CloseNow()
+
+	assertPing(ctx, t, c, bssapi.CmdPingRequest)
+
+	bws := &bssWs{
+		conn: protocol.NewWSConn(c),
+	}
+
+	serializedL2Keystone := hemi.L2KeystoneAbbreviate(includedL2Keystone).Serialize()
+
+	popPayoutsRequest := bssapi.PopPayoutsRequest{
+		L2BlockForPayout: serializedL2Keystone[:],
+	}
+
+	if err := bssapi.Write(
+		ctx, bws.conn, "someid", popPayoutsRequest,
+	); err != nil {
```

Comment:
nit: I think this could be easier to read if it wasn't nested (and at lines 2388-2390, 2415-2417). What are your thoughts on this?

Reply:
I am indifferent 🤷 I can un-nest